File: app/src/main/java/com/qwert2603/spenddemo/changes_list/ChangesListFragment.java
package com.qwert2603.spenddemo.changes_list;

import android.app.Fragment;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.design.widget.Snackbar;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.qwert2603.spenddemo.R;
import com.qwert2603.spenddemo.changes_list.adapter.ChangesAdapter;
import com.qwert2603.spenddemo.data_manager.DataManager;

import butterknife.BindView;
import butterknife.ButterKnife;
import io.reactivex.disposables.CompositeDisposable;
import io.reactivex.disposables.Disposable;

public class ChangesListFragment extends Fragment {

    @BindView(R.id.recycler_view)
    RecyclerView mRecyclerView;

    private DataManager mDataManager = new DataManager();

    private CompositeDisposable mCompositeDisposable = new CompositeDisposable();

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_changes_list, container, false);
        ButterKnife.bind(ChangesListFragment.this, view);

        mRecyclerView.setLayoutManager(new LinearLayoutManager(getActivity()));
        Disposable disposable = mDataManager.getAllChanges()
                .subscribe(changes -> {
                            if (changes.isEmpty()) {
                                View view1 = getView();
                                if (view1 != null) {
                                    Snackbar.make(view1, R.string.no_local_changes, Snackbar.LENGTH_LONG).show();
                                }
                            } else {
                                mRecyclerView.setAdapter(new ChangesAdapter(getActivity(), changes));
                            }
                        },
                        throwable -> {
                            View view1 = getView();
                            if (view1 != null) {
                                Snackbar.make(view1, throwable.toString(), Snackbar.LENGTH_LONG).show();
                            }
                        });
        mCompositeDisposable.add(disposable);

        return view;
    }

    @Override
    public void onDestroyView() {
        mCompositeDisposable.dispose();
        mCompositeDisposable = new CompositeDisposable();
        super.onDestroyView();
    }
}

File: spend/src/main/java/com/qwert2603/spenddemo/changes_list/ChangesListFragment.kt
package com.qwert2603.spenddemo.changes_list

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.qwert2603.spenddemo.R
import com.qwert2603.spenddemo.base_mvi.load_refresh.LoadRefreshPanel
import com.qwert2603.spenddemo.base_mvi.load_refresh.list.ListFragment
import com.qwert2603.spenddemo.di.DIHolder
import com.qwert2603.spenddemo.model.entity.Change
import kotlinx.android.synthetic.main.fragment_changes_list.*
import kotlinx.android.synthetic.main.include_list.*
import kotlinx.android.synthetic.main.toolbar_default.*

class ChangesListFragment : ListFragment<ChangesListModel, ChangesListView, ChangesListPresenter, Change>(), ChangesListView {

    companion object {
        const val TAG = "changes_list"
    }

    override fun loadRefreshPanel(): LoadRefreshPanel = changesList_LRPanelImpl

    override val adapter = ChangesAdapter()

    override fun createPresenter() = DIHolder.diManager.presentersCreatorComponent
            .changesListPresenterComponentBuilder()
            .build()
            .createRecordsListPresenter()

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View
            = inflater.inflate(R.layout.fragment_changes_list, container, false)

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        toolbar.setTitle(R.string.local_changes)
        listEmpty_TextView.setText(R.string.no_local_changes)
    }
}
